File: Код/12.cpp
#include <iostream>
#include <string>

#include <thread>
#include <chrono>


int index;

void write_step(
    int str_i,
    int slag_i,
    const std::string& str,
    const std::string& slag
) {
    std::cout << "\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n";
    for (int j = 0; j < str_i; j++) std::cout << ' ';
    std::cout << "|";
    for (int j = 0; j < slag_i; j++) std::cout << "|";

    std::cout << '\n' << str << "\n----------------------------------------\n";
    std::cout << slag << '\n';
    for (int j = 0; j < slag_i; j++) std::cout << ' ';
    std::cout << "|\n";
    std::this_thread::sleep_for(std::chrono::milliseconds(200));
    
}



bool found(const std::string& str, const std::string& slag) {
    if (slag.size() > str.size()) return false;
    for (size_t str_i = 0; str_i + slag.size() <= str.size(); str_i++) {
        bool is_found = true;
        for (size_t slag_i = 0; slag_i < slag.size(); slag_i++) {
            //write_step(str_i, slag_i, str, slag);
            if (str[str_i + slag_i] != slag[slag_i]) {
                is_found = false;
                break;
            }
        }

        if (is_found) {
            index = str_i;
            return true;
        }
    }

    return false;
}

void replace(std::string& str, const std::string& slag_old, const std::string& slag_new) {
    //std::cout << "> Replacing " << slag_old << " with " << slag_new << '\n';
    if (found(str, slag_old)) {
        str.erase(
            str.begin() + index,
            str.begin() + (int)(index + slag_old.size())
        );
        str.insert(index, slag_new);
    }
    //std::cout << "< finished replacing\n------------\n";
    //std::cout << str << '\n';
}

int main() {
    std::string s = "1";
    for (int i = 0; i < 80; i++) {
        s += '8';
    }
    
    while (found(s, "18") || found(s, "288") || found(s, "3888")) {
        if (found(s, "18")) {
            replace(s, "18", "2");
        } else if (found(s, "288")) {
            replace(s, "288", "3");
        } else if (found(s, "3888")){
            replace(s, "3888", "1");
        }
        //std::cout << s << '\n';
    }
    

    std::cout << "RESULT IS: " << s << '\n';
    return 0;

}

File: Код/12.py
#s = '1' + '8' * 80
#while '18' in s or '288' in s or '3888' in s:
#    if '18' in s:
#        s = s.replace('18','2',1)
#    elif '288' in s:
#        s = s.replace('288','3',1)
#    else:
#        s = s.replace('3888', '1',1)
#print(s)
s = "1" * 78
while "111" in s:
        s = s.replace("111", "2", 1)
        s = s.replace("222", "11", 1)
print(s)

x = '2' + '3' *140
while '2' in x:
        if '23' in x:
                x = x.replace('23','3332',1)
        else:
                x = x.replace('2','333', 1)
print(int(x))

s = '333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333'
print(len(s)*3*3)